### ISSUE TITLE
rmapzen will be broken by geojsonio 0.11.0

### ISSUE THREAD

**Issue by mikemahoney218** [tarakc02/rmapzen#22]
Hi! I'm in the process of submitting geojsonio 0.11.0 to CRAN, which this package depends upon. Tests in this package fail under the new version of geojsonio:

```
Package: rmapzen
Check: tests
New result: ERROR
    Running ‘testthat.R’ [20s/20s]
  Running the tests in ‘tests/testthat.R’ failed.
  Complete output:
    > library(testthat)
    > library(rmapzen)
    > sf::sf_use_s2(TRUE)
    > test_check("rmapzen")
    [ FAIL 1 | WARN 14 | SKIP 2 | PASS 203 ]
    
    ══ Skipped tests ═══════════════════════════════════════════════════════════════
    • skipping multiple contiguous tiles test, requires access to external resource (1)
    • skipping single-tile test, requires access to external resource (1)
    
    ══ Failed tests ════════════════════════════════════════════════════════════════
    ── Error ('test-as-sp.R:21'): as_sp converts vector tile layers to the appropriate Spatial*DF ──
    Error in `maptools::unionSpatialPolygons(res, IDs = res@data$id)`: isTRUE(gpclibPermitStatus()) is not TRUE
    Backtrace:
        ▆
     1. ├─rmapzen::as_sp(ca_tiles$earth, geometry_type = "polygon") at test-as-sp.R:21:4
     2. └─rmapzen:::as_sp.mapzen_vector_layer(ca_tiles$earth, geometry_type = "polygon")
     3.   └─maptools::unionSpatialPolygons(res, IDs = res@data$id)
     4.     └─base::stopifnot(isTRUE(gpclibPermitStatus()))
    
    [ FAIL 1 | WARN 14 | SKIP 2 | PASS 203 ]
    Error: Test failures
    Execution halted
```

The core issue here is that this version of geojsonio drops dependencies on maptools and (more relevant here) rgeos, as part of the [R-spatial evolution project](https://r-spatial.org/r/2022/12/14/evolution2.html). Because geojsonio no longer requires rgeos, rmapzen no longer requires rgeos be installed either. `maptools::unionSpatialPolygons()` requires rgeos for normal functionality; otherwise it can fall back to gpclib, but that package is not permissively licensed, so requires people to "opt-in" to license terms via `gpclibPermit()`.

I believe short term fix would be to include rgeos under Imports in rmapzen. Note though that rgeos and maptools will be [removed from CRAN in October](https://r-spatial.org/r/2022/12/14/evolution2.html), so that would only be a temporary patch; a longer-term fix would be to migrate to sf and related packages.

Let me know if this all makes sense, and if you're able to submit a fixed version of rmapzen to CRAN! If I don't hear back, my current plan is to submit geojsonio 0.11.0 in roughly two weeks, but if you're planning a submission and need a bit more time than that I'm happy to push that back a bit.

**Comment by tarakc02** [comment]
Thanks for this detailed explanation! The package already supports the `sf` types, I had added the `as_sp` functions as an extra, but I can deprecate and remove them without any real hit to the functionality. I'll add `rgeos` to the imports and deprecate `as_sp` for now, and plan to remove `as_sp` entirely in an upcoming release.

**Comment by tarakc02** [comment]
Update: I was able to replicate the described error by installing the development version of geojsonio (current head on github), uninstalling rgeos, reinstalling rmapzen, and running the rmapzen tests. I then implemented the described remedies -- I deprecated the `as_sp` functions and, more immediately importantly, added `rgeos` to rmapzen's imports. After that, installing rmapzen installed rgeos, and the tests were back to all passing locally with the devel version of geojsonio installed. I submitted to CRAN and version 0.4.4 of rmapzen is now live.

I'm keeping this  issue open because, as described above, rmapzen will eventually remove (and not just deprecate) the `as_sp` function, which is the only part of the code that uses rgeos/maptools, in favor of the `as_sf` function.
